framework/generated/dx12_generators/dx12_struct_unwrappers_header_generator.py

```diff
@@ -0,0 +1,195 @@
+        self.newline()
+        write('template <typename T>', file=self.outFile)
+        write(
+            'const T* UnwrapStructPtrObjects(const T* value, HandleUnwrapMemory* unwrap_memory)',
+            file=self.outFile
+        )
+        write('{', file=self.outFile)
+        write('    T* unwrapped_struct = nullptr;', file=self.outFile)
+        self.newline()
+        write('    if (value != nullptr)', file=self.outFile)
+        write('    {', file=self.outFile)
+        write(
+            '        unwrapped_struct = MakeUnwrapStructs(value, 1, unwrap_memory);',
+            file=self.outFile
+        )
+        write(
+            '        UnwrapStructObjects(unwrapped_struct, unwrap_memory);',
+            file=self.outFile
+        )
+        write('    }', file=self.outFile)
+        self.newline()
+        write('    return unwrapped_struct;', file=self.outFile)
+        write('}', file=self.outFile)
+        self.newline()
+        write('template <typename T>', file=self.outFile)
+        write(
+            'const T* UnwrapStructArrayObjects(const T* values, size_t len, HandleUnwrapMemory* unwrap_memory)',
+            file=self.outFile
+        )
+        write('{', file=self.outFile)
+        write('    if ((values != nullptr) && (len > 0))', file=self.outFile)
+        write('    {', file=self.outFile)
+        write(
+            '        auto unwrapped_structs = MakeUnwrapStructs(values, len, unwrap_memory);',
+            file=self.outFile
+        )
+        self.newline()
+        write('        for (size_t i = 0; i < len; ++i)', file=self.outFile)
+        write('        {', file=self.outFile)
+        write(
+            '            UnwrapStructObjects(&unwrapped_structs[i], unwrap_memory);',
+            file=self.outFile
+        )
+        write('        }', file=self.outFile)
+        self.newline()
+        write('        return unwrapped_structs;', file=self.outFile)
+        write('    }', file=self.outFile)
+        self.newline()
+        write(
+            '    // Leave the original memory in place when the pointer is not null, but size is zero.',
+            file=self.outFile
+        )
+        write('    return values;', file=self.outFile)
+        write('}', file=self.outFile)
+        self.newline()
+
+        write('GFXRECON_END_NAMESPACE(encode)', file=self.outFile)
+        write('GFXRECON_END_NAMESPACE(gfxrecon)', file=self.outFile)
+
+        # Finish processing in superclass
+        Dx12BaseGenerator.endFile(self)
+
+    # Method override
+    def generate_feature(self):
+        Dx12BaseGenerator.generate_feature(self)
+
+    def write_struct_unwrap_def(self, struct, indent=''):
+        expr = indent + 'void UnwrapStructObjects({}* value, '\
+            'HandleUnwrapMemory* unwrap_memory);\n'.format(struct)
+
+        write(expr, file=self.outFile)
+
+    def write_include(self):
+        code = ''
+
+        code += '#include "encode/handle_unwrap_memory.h"\n'
```

Comment:
Could add:
```c++
code += '#include "encode/custom_dx12_struct_unwrappers.h"\n'
```
That would reflect that some instantiations of the templates in this generated header dispatch to custom functions in the header included by this line and rely on compilation units which include the generated header also including the custom header.
I'm not pushing strongly for this in case there are places this can be included that never instantiate the templates with types which require the custom header, in which case pulling in an extra header would be bloat.